### ISSUE TITLE
bad path when creating folders for caching search page results

### ISSUE THREAD

**Issue by KY4U** [wp-media/wp-rocket#750]
In the file wp-rocket/inc/front/process.php the _request_uri_path_ is used to create a folder to store the cache pages.   When using the helper plugin **WP Rocket | Cache Search Results**, URLs with the _/?s=something_ parameter are supposed to get cached.  

For example, if a user searches for 'Harmful Algal Blooms', the URL would be something similar to this:
https://mywebsite.com/?s=Harmful+Algal+Blooms

The process.php file tries to create a cache folder at D:\home\site\wwwroot/wp-content/cache/wp-rocket/mywebsite.com/**?s=Harmful+Algal+Blooms** (fails)

Then it tries to write to D:\home\site\wwwroot/wp-content/cache/wp-rocket/mywebsite.com/**?s=Harmful+Algal+Blooms**/index-https.html (fails)

**When the search value is multiple key words separated by the plug sign, the directory creation fails so the cache file is not created.**  

Searches with a single key word, or multiple key words separated by a dash, then the cache file is created and the process flows like this:

For example, if a user searches for 'Harmful-Algal-Blooms' (note the dashes), the URL would be something similar to this:
https://mywebsite.com/?s=Harmful-Algal-Blooms

create cache folder: D:\home\site\wwwroot/wp-content/cache/wp-rocket/mywebsite.com/?s=harmful-algal-blooms (fails)

saving to: D:\home\site\wwwroot/wp-content/cache/wp-rocket/mywebsite.com/?s=harmful-algal-blooms/index-https.html (fails)

create cache folder: D:\home\site\wwwroot/wp-content/cache/wp-rocket/mywebsite.com/search/harmful-algal-blooms/page/2 (successful)

saving to: D:\home\site\wwwroot/wp-content/cache/wp-rocket/mywebsite.com/search/harmful-algal-blooms/page/2/index-https.html (successful)

Bottom line is this:  searches with a single keyword, or multiple keywords separated with dashes, will result in a cache page being created.  Searches with multiple keywords separated with plus signs will result in cache pages **not** being created.

**Comment by Tabrisrp** [comment]
Thank you for the complete report, we will look into this to improve the caching of search results

**Comment by Screenfeed** [comment]
@KY4U Are you still experiencing the issue? I couldn't reproduce it myself, all seems to work perfectly.

**Comment by KY4U** [comment]
@Screenfeed Having the ability to cache search result pages was a critical feature I needed.  Unfortunately the errors that I reported (on a Windows system) prevented me from using this plugin and I had to move on.

**Comment by Screenfeed** [comment]
@KY4U OK, thanks for your reply :)
(now that means you can come back since it seems to work 😄)
